kitsune/wiki/utils.py

```diff
@@ -198,3 +200,36 @@ def get_visible_document_or_404(
 
 def get_visible_revision_or_404(user, **kwargs):
     return get_object_or_404(Revision.objects.visible(user, **kwargs))
+
+
+def build_topics_data(request, product, topics):
+    """Build topics_data for use in topic cards"""
+    topics_data = []
+    for topic in topics:
+        # Get featured articles first (returns up to 4)
+        featured_articles = get_featured_articles(
+            product, locale=request.LANGUAGE_CODE, topic=topic
+        )
+
+        # Always get total count of available documents
+        docs, _ = documents_for(
```

Comment:
Ideally this should be moved outside of the for loop by fetching documents for all topics and then organize them per topic. This way you can loop over data in memory instead of querying the database on each loop iteration